frame/multi-account/src/lib.rs

```diff
@@ -0,0 +1,1506 @@
+		/// The signatories were provided out of order; they should be ordered.
+		SignatoriesOutOfOrder,
+		/// The sender was contained in the other signatories; it shouldn't be.
+		SenderInSignatories,
+		/// Multi account not found.
+		MultiAccountNotFound,
+		/// Multisig operation not found when attempting to cancel.
+		NotFound,
+		/// Only the account that originally created the multisig is able to cancel it.
+		NotOwner,
+		/// The sender is not a signatory of this multi account
+		NotSignatory,
+		/// No timepoint was given, yet the multisig operation is already underway.
+		NoTimepoint,
+		/// A different timepoint was given to the multisig operation that is underway.
+		WrongTimepoint,
+		/// A timepoint was given, yet no multisig operation is underway.
+		UnexpectedTimepoint,
+	}
+}
+
+decl_event! {
+	/// Events type.
+	pub enum Event<T> where
+		AccountId = <T as system::Trait>::AccountId,
+		BlockNumber = <T as system::Trait>::BlockNumber
+	{
+		/// A multi account has been created. First param is the account that created it, second is the multisig
+		/// account.
+		NewMultiAccount(AccountId, AccountId),
+		/// A multi account has been updated. First param is the multisig account.
+		MultiAccountUpdated(AccountId),
+		/// A multi account has been removed. First param is the multisig account.
+		MultiAccountRemoved(AccountId),
+		/// A new multisig operation has begun. First param is the account that is approving,
+		/// second is the multisig account.
+		NewMultisig(AccountId, AccountId),
+		/// A multisig operation has been approved by someone. First param is the account that is
+		/// approving, third is the multisig account.
+		MultisigApproval(AccountId, Timepoint<BlockNumber>, AccountId),
+		/// A multisig operation has been executed. First param is the account that is
+		/// approving, third is the multisig account.
+		MultisigExecuted(AccountId, Timepoint<BlockNumber>, AccountId, DispatchResult),
+		/// A multisig operation has been cancelled. First param is the account that is
+		/// cancelling, third is the multisig account.
+		MultisigCancelled(AccountId, Timepoint<BlockNumber>, AccountId),
+	}
+}
+
+/// Simple index-based pass through for the weight functions.
+struct MultiPassthrough<Call, AccountId, Timepoint>(
+	sp_std::marker::PhantomData<(Call, AccountId, Timepoint)>
+);
+
+impl<Call, AccountId, Timepoint> MultiPassthrough<Call, AccountId, Timepoint> {
+	fn new() -> Self { Self(Default::default()) }
+}
+impl<Call: GetDispatchInfo, AccountId, Timepoint> WeighData<(&AccountId, &Timepoint, &Box<Call>)>
+for MultiPassthrough<Call, AccountId, Timepoint>
+{
+	fn weigh_data(&self, (_, _, call): (&AccountId, &Timepoint, &Box<Call>)) -> Weight {
+		call.get_dispatch_info().weight
+	}
+}
+impl<Call: GetDispatchInfo, AccountId, Timepoint> ClassifyDispatch<(&AccountId, &Timepoint, &Box<Call>)>
+for MultiPassthrough<Call, AccountId, Timepoint>
+{
+	fn classify_dispatch(&self, (_, _, call): (&AccountId, &Timepoint, &Box<Call>))
```

Comment:
All of this passthrough stuff can be updated with `FunctionOf` in the weight. You can check out master for the latest here.

Reply:
Awesome, changed. Thanks for pointing this out!